File: Linear/Static/Array/03_Deleting/Given_Position/Custom/Deleting.c
#include <stdio.h>

void deleteAtGivenPosition(int arr[], int n, int pos) {
    // Delete the element at the given position.
    for (int i = pos; i < n; i++) {
        arr[i - 1] = arr[i];
    }
}

int main() {
    int arr[] = {10, 20, 30, 40};
    int n = sizeof(arr) / sizeof(arr[0]);
    
    int pos;

    printf("Array before deletion:\n");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

    printf("\nEnter a position: ");
    scanf("%d", &pos);

    deleteAtGivenPosition(arr, n, pos);
    n--;

    printf("\nArray after deletion:\n");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    
    return 0;
}

File: Linear/Static/Array/03_Deleting/Given_Position/Custom/Deleting.java
import java.util.Scanner;

class DeletingGivenPositionCustom {
    public static int[] deleteAtGivenPosition(int[] arr, int pos) {
        int[] newArr = new int[arr.length - 1];

        for (int i = 0; i < pos - 1; i++) {
            newArr[i] = arr[i];
        }

        // Delete the element at the given position.
        for (int i = pos; i < arr.length; i++) {
            newArr[i - 1] = arr[i];
        }

        return newArr;
    }
    
    public static void main(String[] args) {
        int[] arr = {10, 20, 30, 40};
        
        System.out.println("Array before deletion");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }

        Scanner sc = new Scanner(System.in);
        System.out.println("\nEnter a position: ");
        int pos = sc.nextInt();

        int[] newArr = deleteAtGivenPosition(arr, pos);

        sc.close();

        System.out.println("\nArray after deletion");
        for (int i = 0; i < newArr.length; i++) {
            System.out.print(newArr[i] + " ");
        }
    }
}